### PR TITLE
Add `--icon-spacing` flag and `icon.spacing` config option

### PR DESCRIPTION
**Pull request by OrangeFran**
The spacing option takes a string and puts it between the icon and the name. The default would be one space: " ". Requested feature from the following issue: #363.

Example:

``` sh
lsd --icon-spacing "  " # I removed this now
```
``` yaml
icons:
    separator: "  "
```

Edit: I don't really know what `codecov/project` does and why it fails. Please explain.

---
#### TODO

- [x] Use `cargo fmt`
- [x] Add necessary tests (I hope the updated tests are sufficient)
- [x] Add changelog entry

### CODE REVIEW

**Comment by zwpaper** [comment]
hi @OrangeFran , thanks for the PR, for the `codecov/project` part, you can click the `Detail` on the right side, it shows that you changed the `src/flags/icons.rs` and make the coverage go down 0.07, you should add more test for it.

image

**Comment by meain** [comment]
I am thinking we could give the icon spacing option just in the cofig file and not in cli options as this is will be changed that often. Also, we could read the icon spacing value as an int rather than a string of spaces.

**Comment by OrangeFran** [comment]
The cli option could be removed, I agree with you here.

I don't think that we should replace the string with an int. The string is just more customizable. I might use ` |` or something like that in the future. So I'd like to keep the string if possible.

**Comment by meain** [comment]
Ahh, I see your point. In that case I think we should probably call it icon.separator instead of spacing.

**Comment by OrangeFran** [comment]
Oh, I spelled `separator` wrong. Let me change that. Sorry.

File: src/icon.rs [hunk 1]
```diff
@@ -370,10 +370,10 @@ mod test {
         File::create(&file_path).expect("failed to create file");
         let meta = Meta::from_path(&file_path, false).unwrap();
 
-        let icon = Icons::new(Theme::Fancy);
-        let icon = icon.get(&meta.name);
+        let icon = Icons::new(Theme::Fancy, " ".to_string());
+        let icon_str = icon.get(&meta.name);
 
-        assert_eq!(icon, format!("{}{}", "\u{f016}", ICON_SPACE)); // 
+        assert_eq!(icon_str, format!("{}{}", "\u{f016}", icon.icon_separator)); // 
```
Comment (meain) [inline src/icon.rs:376]: Can we also add a test for a non default icon separator is being used.

**Review by meain — COMMENTED** [review]
Just a few more things

File: src/app.rs [hunk 2]
```diff
@@ -53,6 +53,13 @@ pub fn build() -> App<'static, 'static> {
                 .number_of_values(1)
                 .help("Whether to use fancy or unicode icons"),
         )
+        // .arg(
```
[the remark this answers is not in the document]
Comment (OrangeFran) [reply src/app.rs:56]: Sorry, I forgot to delete them.

File: src/config_file.rs [hunk 3]
```diff
@@ -212,6 +213,9 @@ icons:
   # Which icon theme to use.
   # Possible values: fancy, unicode
   theme: fancy
+  # The string between the icons and the name.
+  # Possible values: any string
```
[the remark this answers is not in the document]
Comment (OrangeFran) [reply src/config_file.rs:217]: Makes sense.

File: src/icon.rs (hunk 1 above)
Comment (OrangeFran) [reply src/icon.rs:376]: Should I put the test there (`src/icon.rs`) or in the `src/flags/icon.rs` file, where I already created a test for the default case?

Comment (meain) [reply src/icon.rs:376]: I think we will keep it in `src/flags/icon.rs`.

Comment (OrangeFran) [reply src/icon.rs:376]: I made the changes.  You can review again now.

**Comment by meain** [comment]
Looks good! 🎉

**Pull request merged**